server/src/main/java/org/candlepin/controller/CandlepinOwnerProductShareManager.java

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright (c) 2009 - 2017 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.controller;
+
+import org.candlepin.model.Owner;
+import org.candlepin.model.OwnerProductCurator;
+import org.candlepin.model.OwnerProductShare;
+import org.candlepin.model.OwnerProductShareCurator;
+import org.candlepin.model.Product;
+
+import com.google.inject.Inject;
+
+import org.apache.commons.collections.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Manages Shared product operations.
+ */
+public class CandlepinOwnerProductShareManager implements OwnerProductShareManager {
```

Comment:
Is there (or do we anticipate there will be) more than one implementation of the OwnerProductShareManager interface?  I'm guessing that there's just going to be the one implementation so unless we need an interface for testing purposes or something, I don't see the advantage in having an interface.